scripts/hudson/narayana.sh

```diff
@@ -411,20 +411,27 @@ function clone_as {
 }
 
 function build_as {
-  echo "Building JBoss EAP/WildFly"
+  REPO="https://github.com/wildfly/wildfly.git"
+  AS_BRANCH=27.0.0.Alpha1
+  echo "Building WildFly $AS_BRANCH"
 
-  cd $WORKSPACE/jboss-as
+  cd $WORKSPACE
+  if [ -d jboss-as ]; then
+    rm -rf jboss-as # start afresh
+  fi
+  git clone $REPO jboss-as || fatal "git clone $REPO failed"
+  cd jboss-as
+
+  git checkout $AS_BRANCH || fatal "git checkout $AS_BRANCH failed"
 
   # building WildFly
   [ "$_jdk" -lt 17 ] && export MAVEN_OPTS="-XX:MaxMetaspaceSize=512m -XX:+UseConcMarkSweepGC $MAVEN_OPTS"
   [ "$_jdk" -ge 17 ] && export MAVEN_OPTS="-XX:MaxMetaspaceSize=512m $MAVEN_OPTS"
-  JAVA_OPTS="-Xms1303m -Xmx1303m -XX:MaxMetaspaceSize=512m $JAVA_OPTS" ./build.sh clean install -B -DskipTests -Dts.smoke=false $IPV6_OPTS -Dversion.org.jboss.narayana=${NARAYANA_CURRENT_VERSION} "$@"
-  [ $? -eq 0 ] || fatal "AS build failed"
+  JAVA_OPTS="-Xms1303m -Xmx1303m -XX:MaxMetaspaceSize=512m $JAVA_OPTS" ./build.sh clean install -B -DskipTests -Dts.smoke=false $IPV6_OPTS -Dversion.org.jboss.narayana=${NARAYANA_CURRENT_VERSION} "$@" || fatal "AS build failed"
```

Comment:
Is this tidy up?

Reply:
It is a simplification to reduce the cognitive overload, if that's clean up then yes.

Reply:
We use the pattern `[ $? -eq 0 ] || fatal` many places in https://github.com/jbosstm/narayana/blob/master/scripts/hudson/narayana.sh. If a change is necessary, I think they should all be proposed to be changed at once but I would suggest to discuss the change on Zulip and give time to receive feedback.

Reply:
I will remove them. But discussing on Zulip is overkill, micro-managing software updates seems wrong and will turn people off from reading our chat so let's limit such discussions to the PR's.

But, at risk of repeating myself, anything we can do to simplify this behemoth of a script should be encouraged, so when I find the time I will raise a PR to simplify it - we have all been guilty of misunderstanding how this script works.